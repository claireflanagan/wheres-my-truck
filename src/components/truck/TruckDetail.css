.TruckDetail {
  margin: 1em;
  padding: 1em;
  background: rgba(255,255,255, 0.8);

  @media screen and (min-width: 1200px) {
    max-width: 75em;
    margin: auto;
  }

  dt {
    margin: 3px;
  }

  dd {
    margin-left: 10px;
  }

  span {
    font-weight: bold;
  }

  header {
    display: flex;
    justify-content: space-between;
    margin-bottom: 1em;

    h2 {
      margin: 0;
    }

    h3 {
      margin: 0;
      font-size: 1em;
    }

    .use {
      display: inline-block;
      max-height: 30px;
      margin: 1em;
      padding: 0.4em;
      border-radius: 0.4em;
      color: #FFFFFF;
      font-size: 1.4rem;
      vertical-align: top;
    }

    .free {
      background-color: #679D69;
    }

    .ride {
      background-color: #679b9d;
    }
  }

  .truckLinks {
    display: flex;
    flex-direction: column;

    @media screen and (min-width: 900px) {
      flex-direction: row;
    }

    a {
      display: block;
      width: 100%;
      margin: 0.6rem 0;
      padding: 1em;
      border: 3px solid #679D69;
      border-radius: 0.2em;
      color: black;

      font-size: 1.4em;
      text-decoration: none;
      text-align: center;
      &:hover {
        background: white;
      }

      @media screen and (min-width: 900px) {
        margin: 1rem 0.6rem;
      }
    }
  }


}
