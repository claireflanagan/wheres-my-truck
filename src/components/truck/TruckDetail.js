import React, { Component } from 'react';
import styles from './TruckDetail.css';
import { ROUTES } from '../../routes/routes';
import { Link } from 'react-router-dom';

  

export default class TruckDetail extends Component { 
  componentDidMount() {
    this.props.getTruck(this.props.match.params.id);
  }
  render() {
    const { truck } = this.props;
    console.log('truck4354', truck);
    return (
      <div>
        <div className={styles.truckLinks}>
          <Link to={ROUTES.IMAGEDISPLAY.linkTo(truck.id, 'registration')}>Registration</Link>
          <Link to={ROUTES.IMAGEDISPLAY.linkTo(truck.id, 'insurance')}>Insurance</Link>
          <Link to={ROUTES.MAINTENANCE_LIST.linkTo(truck.id)}>Maintenance Records</Link>
        </div>
        <table className={styles.table}>
          <thead> 
            <tr>
              <th colSpan="2">{truck.name}</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Year:</td>
              <td>{truck.year}</td>
            </tr>
            <tr>
              <td>Make:</td>
              <td>{truck.make}</td>
            </tr>
            <tr>
              <td>Model:</td>
              <td>{truck.model}</td>
            </tr>
            <tr>
              <td>Vin:</td>
              <td>{truck.vin}</td>
            </tr>
            <tr>
              <td>Plates:</td>
              <td>{truck.plates}</td>
            </tr>
            <tr>
              <td>Tire Size:</td>
              <td>{truck.tireSize}</td>
            </tr>
            <tr>
              <td>Year Bought:</td>
              <td>{truck.boughtDate}</td>
            </tr>
          </tbody>
        </table>
      </div>
    );
  }
}


