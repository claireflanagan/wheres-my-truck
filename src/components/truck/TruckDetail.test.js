import React from 'react';
import { shallow } from 'enzyme';
import TruckDetail from './TruckDetail';

jest.mock('../../routes/routes');
jest.mock('../../services/auth.js');

describe('TruckDetail component', () => {
  const truck =
  {
    id: 'a0',
    name: 'Truck 0',
    location: '10th and Couch',
    vin: '1FTFS24Y8PHA25143',
    plates: '876ABC',
    year: 2018,
    make: 'Truck',
    model: 'Big Truck',
    tireSize: 20,
    boughtDate: '2017-09-10',
    registration: 'https://www.dmv.ca.gov/imageserver/dmv/images/vr/regcard_w_arrow2.jpg',
    insurance: 'https://approvedauto.files.wordpress.com/2013/12/id-card-example.jpg'
  };
  const match = { params: { id: 'a0' } };
  it('matches a snapshot', () => {
    expect(true);
    const wrapper = shallow(<TruckDetail
      truck={truck}
      match={match}
      getTruck={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
