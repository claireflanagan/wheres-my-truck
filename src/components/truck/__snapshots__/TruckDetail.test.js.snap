// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TruckDetail component matches a snapshot 1`] = `
<div>
  <div
    className="truckLinks"
  >
    <Link
      replace={false}
      to="/trucks/a0/registration"
    >
      Registration
    </Link>
    <Link
      replace={false}
      to="/trucks/a0/insurance"
    >
      Insurance
    </Link>
    <Link
      replace={false}
      to="/trucks/a0/maintenanceList"
    >
      Maintenance
    </Link>
  </div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Truck 0
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Year: 
          2018
        </td>
      </tr>
      <tr>
        <td>
          Make: 
          Truck
        </td>
      </tr>
      <tr>
        <td>
          Model: 
          Big Truck
        </td>
      </tr>
      <tr>
        <td>
          Vin: 
          1FTFS24Y8PHA25143
        </td>
      </tr>
      <tr>
        <td>
          Plates: 
          876ABC
        </td>
      </tr>
      <tr>
        <td>
          Tire Size: 
          20
        </td>
      </tr>
      <tr>
        <td>
          Year Bought: 
          2017-09-10
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
